internal/plugin/config_plugin_test.go

```diff
@@ -152,10 +151,124 @@ func TestConfig_Process(t *testing.T) {
+	instance := instances.Instance{
+		InstanceId: instanceID,
+		Type:       instances.Type_NGINX,
+	}
+
+	location := fmt.Sprintf("/instance/%s/files/", instanceID)
+	request := model.InstanceConfigUpdateRequest{
+		Instance:      &instance,
+		Location:      location,
+		CorrelationID: correlationID,
+	}
+
+	inProgressStatus := helpers.CreateInProgressStatus()
+	successStatus := helpers.CreateSuccessStatus()
+	failStatus := helpers.CreateFailStatus("error")
+	rollbackInProgressStatus := helpers.CreateRollbackInProgressStatus()
+
+	tests := []struct {
+		name               string
+		updateReturnStatus *instances.ConfigurationStatus
+		rollbackReturns    error
+		expected           []*bus.Message
+	}{
+		{
+			name:               "success",
+			updateReturnStatus: successStatus,
+			rollbackReturns:    nil,
+			expected: []*bus.Message{
+				{
+					Topic: bus.InstanceConfigUpdateTopic,
+					Data:  inProgressStatus,
+				},
+				{
+					Topic: bus.InstanceConfigUpdateTopic,
+					Data:  successStatus,
+				},
+			},
+		},
+		{
+			name:               "fail and rollback",
+			updateReturnStatus: failStatus,
+			rollbackReturns:    nil,
+			expected: []*bus.Message{
+				{
+					Topic: bus.InstanceConfigUpdateTopic,
+					Data:  inProgressStatus,
+				},
+				{
+					Topic: bus.InstanceConfigUpdateTopic,
+					Data:  failStatus,
+				},
+				{
+					Topic: bus.InstanceConfigUpdateTopic,
+					Data:  rollbackInProgressStatus,
+				},
+				{
+					Topic: bus.InstanceConfigUpdateTopic,
+					Data:  helpers.CreateRollbackSuccessStatus(),
+				},
+			},
+		},
+		{
+			name:               "rollback fails",
+			updateReturnStatus: failStatus,
+			rollbackReturns:    fmt.Errorf("rollback failed"),
+			expected: []*bus.Message{
+				{
+					Topic: bus.InstanceConfigUpdateTopic,
+					Data:  inProgressStatus,
+				},
+				{
+					Topic: bus.InstanceConfigUpdateTopic,
+					Data:  failStatus,
+				},
+				{
+					Topic: bus.InstanceConfigUpdateTopic,
+					Data:  rollbackInProgressStatus,
+				},
+				{
+					Topic: bus.InstanceConfigUpdateTopic,
+					Data:  helpers.CreateRollbackFailStatus("rollback failed"),
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(tt *testing.T) {
+			messagePipe := bus.NewFakeMessagePipe(context.TODO())
+			configPlugin := NewConfig(&agentConfig)
+
+			err := messagePipe.Register(10, []bus.Plugin{configPlugin})
+			require.NoError(tt, err)
+			messagePipe.Run()
+
+			configService := &servicefakes.FakeConfigServiceInterface{}
+			configService.UpdateInstanceConfigurationReturns(make(map[string]*instances.File), test.updateReturnStatus)
+			configService.RollbackReturns(test.rollbackReturns)
+
+			instanceService := []*instances.Instance{&instance}
+			configPlugin.configServices[instanceID] = configService
 			configPlugin.instances = instanceService
 
-			configPlugin.Process(test.input)
+			configPlugin.updateInstanceConfig(&request)
 
 			messages := messagePipe.GetMessages()
 
```

Comment:
Can you validate each message as well instead of just validating that we get the correct number of them?